scipy/stats/stats.py

```diff
@@ -5718,3 +5720,175 @@ def rankdata(a, method='average'):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+def ratio_unif(f, size=1, x1=None, x2=None, y2=None, shift=0, maxiter=None,
+               bounds=None):
+    """
+    Generate random variables using ratio-of-uniforms method.
+
+    Given a continuous univariate density f and a constant c, define the set
+    A(c) := {(x, y) : 0 < y <= sqrt(f(x/y + c)) }. If (U, V) is a random
+    vector uniformly distributed over A(c), then U/V + c follows a distribution
+    with density f.
+
+    The above result (see [1]_, [2]_) can be used to sample random variables
+    using only the density. Typical choices of c are zero or the mode of f.
```

Comment:
Might want to outline the circumstances for when it's going to be better/faster to use this function than calculate via pdf-->cdf-->ppf-->rvs.